extensions/livedata/src/main/java/com/twocoders/extensions/livedata/LiveDataExtensions.kt

```diff
@@ -0,0 +1,382 @@
+package com.twocoders.extensions.livedata
+
+import androidx.annotation.MainThread
+import androidx.lifecycle.*
+import com.hadilq.liveevent.LiveEvent
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+
+fun <T : Any?> LiveData<T>.asLiveEvent(): LiveEvent<T> =
+    if (this is LiveEvent<T>) this else throw IllegalArgumentException("$this is not an instance of LiveEvent!")
+
+fun <T : Any?> LiveData<T>.asMutable(): MutableLiveData<T> =
+    if (this is MutableLiveData<T>) this else throw IllegalArgumentException("$this is not an instance of MutableLiveData!")
+
+fun LiveData<String>.isEmpty(): Boolean = value?.isEmpty() ?: true
+
+fun LiveData<String>.isNotEmpty() = !isEmpty()
+
+fun LiveData<Boolean>.isTrue() = value == true
+
+fun LiveData<Boolean>.isNotTrue() = !isTrue()
+
+fun MutableLiveData<Boolean>.toggle() {
+    value = !value!!
+}
+
+fun <T : Any?> MutableLiveData<T>.notifyObservers() {
+    this.value = this.value
+}
+
+fun <K, V> MutableLiveData<Map<K, V>>.put(key: K, newValue: V) {
```

Comment:
technically this should be `MutableLiveData<MutableMap<K, V>>` or some check preventing casting `Map` to `MutableMap`